frontend/public/components/app.jsx

```diff
@@ -129,9 +129,15 @@ class App extends React.PureComponent {
         <GlobalNotifications />
         <Switch>
           <Route path={['/all-namespaces', '/ns/:ns',]} component={RedirectComponent} />
+
+          <LazyRoute path="/catalog/all-namespaces" exact loader={() => import('./catalog' /* webpackChunkName: "catalog" */).then(m => m.CatalogPage)} />
+          <LazyRoute path="/catalog/ns/:ns" exact loader={() => import('./catalog' /* webpackChunkName: "catalog" */).then(m => m.CatalogPage)} />
```

Comment:
Really concerned about adding a namespaced route that is not a real Kubernetes resource view. Looking at the `<Firehose>` components that you use, the data does not come from the current namespace. Wouldn't it make more sense to have the be a non-namespaced route (`/catalog`) and then have the user choose which namespace to add the catalog item to later?

Reply:
We need also to get image streams and service classes from the current namespace.

We do this for status already fwiw

Reply:
Looks like `ImageStreams` are fetched from[ the `openshift` namespace](https://github.com/openshift/console/pull/530/files#diff-0a92d8d0e684429073eba8d7b8b34ecfR165).

That's true, we have some non-k8s namespaced routes, but [none of them show a list of resources](https://github.com/openshift/console/pull/530/files#diff-0a92d8d0e684429073eba8d7b8b34ecfR165) (other than search). This causes deviation from the `kubectl` experience (I'm viewing `Deployments` in `my-ns`; I can `kubectl get -n my-ns deployments`).

Reply:
But we will also need to eventually get image streams and service classes from the current namespace. (It's just not implemented yet.)

In console, there is search, status, and overview, which all show resources. (Status shows events.)

There is also `oc get all` and `oc status`. I'm not sure this really deviates from the CLI experience.

We didn't namespace the catalog in the previous implementation, and it caused a lot of headaches for us later.

Reply:
@alecmerdler I thought the namespace would be used for what @spadgett mentioned in https://jira.coreos.com/browse/CONSOLE-694: 

"There is a namespaced ServiceClass resource as well, but we haven't yet added support for it. Ideally we'd also get ImageStreams that are in the current namespace."